src/algebraic_geometry/morphisms/basic.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import algebraic_geometry.AffineScheme
+import algebraic_geometry.pullbacks
+
+/-!
+# Properties of morphisms
+
+We provide the basic framework for talking about properties of morphisms.
+The following meta-properties are defined
+
+* `respects_iso`: `P` respects isomorphisms if `P f → P (e ≫ f)` and `P f → P (f ≫ e)`, where
+  `e` is an isomorphism.
+* `stable_under_composition`: `P` is stable under composition if `P f → P g → P (f ≫ g)`.
+* `stable_under_base_change`: `P` is stable under base change if `P (Y ⟶ S) → P (X ×[S] Y ⟶ X)`.
+
+-/
+
+universe u
+
+open topological_space category_theory category_theory.limits opposite
+
+noncomputable theory
+
+namespace algebraic_geometry
+
+/-- A `morphism_property` is a class of morphisms between schemes. -/
+@[derive complete_lattice]
+def morphism_property := ∀ ⦃X Y : Scheme⦄ (f : X ⟶ Y), Prop
```

Comment:
Should the most general version of this be parameterised over the base category? Because this also makes sense in `Top` and `CommRing`.